### ISSUE TITLE
Resource bundles in non-standard location

### ISSUE THREAD

**Issue by jander99** [psi-probe/psi-probe#505]
With a "standard" maven directory structure, message resource bundles typically go in /src/main/resources/ and not /src/main/conf or /src/main/conf-filtered.  The maven-resources-plugin should be able to handle filtering messages. We just need to update the application to look in /src/main/resources instead of /src/main/conf.

Please tag for 3.0.0. For now, l10n-maven-plugin as part of #504 will not pick up /src/main/conf.  This can possibly be passed via a configuration, but the better solution is to move the resource bundles to the "more proper" location.

**Comment by hazendaz** [comment]
PR #517 combines /src/main/conf and /src/main/conf-filtered.  I do agree they are in wrong place but don't really see a point to waiting until 3.0.0 to fix.  So maybe after #517 is on board, we can refactor and fix this.  It should be pretty easy and is just a build item.  In fact, they can stay as is and simply move to resources.  It will copy automatically anyways to the correct location then as it will correctly match the final layout.  Filtering should work just the same.  Had I noticed this first, I think I would have just did this in one go.

**Comment by hazendaz** [comment]
Original issue after testing and verification is incorrect in proper location.  That works for a JAR.  Being this is a WAR, an issue still exists but actually things just need to move under src/main/webapp.  I've done the change and verified it works in tomcat. I did try the original suggestion first with a number of modifications and it would not work.  Things just got moved around too deeply in the end and/or required far more changes than necessary.  The new build though has 1 less file.  Originally the context.xml was getting put in two places.  It stated specifically to put under META-INF for jboss but actually that is the same for jboss and tomcat.  Feel free to correct me if I missed anything and check out the PR.

**Comment by hazendaz** [comment]
PR didn't work out as it is auto changing line endings and various tries failed to get that to stop occurring.  So this will be on hold until after code base is normalized.  Everything works as-is now so this is just a cleanup effort.

**Comment by MALfunction84** [comment]
As far as I can tell, files in src/main/resources end up being copied into WEB-INF/classes when you're dealing with the maven-war-plugin, and no filtering is applied. Some of the resources (messages*.properties and version.properties) need to be copied to WEB-INF and filtered. Others just need to be filtered (log4j.properties). If there is a way to do these without using ```<webResource>```, I don't know of it.

It's true that context.xml is in META-INF by default. The JBoss weirdness in there copies context.xml *from* META-INF *to* WEB-INF. This is documented [here](https://developer.jboss.org/wiki/Web-AppContextConfiguration).

**Comment by hazendaz** [comment]
My change here at this point is not moving anything over to resources.  That didn't work too well.  I've applied what you did with log4j piece and put in struture for war.  So no conf directory.  Items are directly in webapp.  It's much smaller in scope than original issue reported here.  However, due to some line ending issues, it still looks like larger change.  I don't see anyway around that.  Moving the fixes in fact changes the line endings on some that are incorrect.  It's still easy to see not change. I can submit PR again if you want to take a look.

**Comment by MALfunction84** [comment]
Everything in the repo should be UNIX-style line endings. If there are inconsistencies in the repo, maybe we should fix those separately so as not to obfuscate other changes.

#518 included a change to .gitattributes that introduced ```text eol=crlf```, so maybe that was the problem. I've got my git set up to convert to UNIX-style line endings automatically prior to commit.

If we put the resources directly in webapp, how is filtering applied? With the exception of stats.properties, everything in conf is filtered to replace properties with their values.

**Comment by hazendaz** [comment]
I’m pretty sure the entire project currently is not unix style line endings but rather windows styled.  Add a git attributes and watch what happens when files are touched.  Below ensures properly line normalization per github which would mean it’s correct in the REPO and unix style.  When I add this using Github for Windows Client, it changes 232 files as it applies it at once.  What it doesn’t change is all the files within the ‘resource location change’ which my local settings enforced unix style.  I did have a change in #518 trying to mess around with that little bit.  The fact though is that crlf there actually made it apply windows endings which is the only setting that at the moment is correct against this library in mass.  If I switch to that, then it immediately says no changes are necessary.  So definitely the library is set wrong and not normalized across files.  Only way to fix it is to add the below entry in the suggested git attributes default and do a one-time push.  It is certainly a very messy thing to do.  However, given formatting just occurred this is a really good time to apply that.

 

# Auto detect text files and perform LF normalization
* text=auto

 

Applying the above won’t affect me much on either of the outstanding items I have (resources move and spring 3).  Both are easily put back into the flow so non issue but up to you if you really want to fix the code repo.  This is a pretty easily missed item across git projects and the above setting always works well.  If you use github for windows client, it immediate touches all files.  Otherwise, I’ve seen it require a separate touch if it is wanted in one go.  I do believe some files are simply treated differently at times and being that I was trying to move some, that is why it then did a switch.  So in fact, I think if git is setup correctly locally all new files are probably getting applied properly.  Its most like the very old original ones that had these issues.

 

Anyway, to the resource filtering, I left that as is too.  It just filters more items.  All that ultimately does is slow the build ever so slightly so it can determine what is necessary to filter.  I didn’t see any negative outcome in keeping those together.  The original request was to get them under resources but ultimately that didn’t work out well so it was rather just bringing them together that did the trick.  The pom just gets ‘conf’ replaced with ‘webapps’ and nothing more.  The contents of ‘conf’ are just moved into ‘webapps’.  Nothing more than that.  If the line endings are fully addressed, then it will come across properly.

 

Anyway, if you want, I can do a PR to push up the line normalization in case you are not able to replicate it as easily.  Just let me know if you want that and I’ll get it up there.

 

Thanks,

 

Jeremy

**Comment by hazendaz** [comment]
I’m going to go ahead and put up a PR on this.  You can close it out if you want.  It’s more for reference and probably better if you do this yourself so you are completely sure what is happening :)
